Sixteen750/src/main/java/org/firstinspires/ftc/sixteen750/Robot.java

```diff
@@ -1,48 +1,46 @@
 package org.firstinspires.ftc.sixteen750;
 
+import org.firstinspires.ftc.sixteen750.command.autonomous.StartingPosition;
 import org.firstinspires.ftc.sixteen750.subsystem.ArmSubsystem;
 import org.firstinspires.ftc.sixteen750.subsystem.ClawSubsystem;
-import org.firstinspires.ftc.sixteen750.subsystem.MecanumDriveSubsystem;
 import org.firstinspires.ftc.sixteen750.subsystem.LiftSubsystem;
+import org.firstinspires.ftc.sixteen750.subsystem.MecanumDriveSubsystem;
+import org.firstinspires.ftc.sixteen750.subsystem.VisionSubsystem;
 
 import com.acmerobotics.dashboard.config.Config;
 
 import com.technototes.library.logger.Loggable;
+import com.technototes.library.util.Alliance;
 
 public class Robot implements Loggable {
     @Config
     public static class RobotConstant {
-        public static boolean SWERVE_DRIVE_ENABLED = false;
-        public static boolean TANK_DRIVE_ENABLED = false;
-        public static boolean MECANUM_DRIVE_ENABLED = false;
-        public static boolean CLAW_ENABLED = false;
-        public static boolean ARM_ENABLED = false;
+        // The only purpose of RobotConstant is to enable/disable subsystem(s) in FTC-Dashboard for non-testing OpMode
+        private static boolean SWERVE_DRIVE_ENABLED = false;
+        private static boolean TANK_DRIVE_ENABLED = false;
+        public static boolean MECANUM_DRIVE_ENABLED = true;
         public static boolean LIFT_ENABLED = true;
-        public static boolean LIFT_MOVE_MOTORS = true;
+        public static boolean ARM_ENABLED = true;
+        public static boolean CLAW_ENABLED = true;
+        public static boolean CAMERA_ENABLED = true;
     }
 
     public ClawSubsystem clawSubsystem;
     public ArmSubsystem armSubsystem;
     public LiftSubsystem liftSubsystem;
     public MecanumDriveSubsystem mecanumDriveSubsystem;
+    public VisionSubsystem visionSubsystem;
 
-    public boolean isSwerveDriveConnected = false;
-    public boolean isTankDriveConnected = false;
-    public boolean isMecanumDriveConnected = false;
-    public boolean isClawConnected = false;
-    public boolean isArmConnected = false;
-    public boolean isLiftConnected = true;
-
-    public Robot(Hardware hardware, boolean enableMecanumDrive, boolean enableLift, boolean enableArm, boolean enableClaw) {
+    public Robot(Hardware hardware, boolean enableMecanumDrive, boolean enableLift, boolean enableArm, boolean enableClaw, boolean enableCamera, Alliance alliance, StartingPosition whichSide) {
         if (enableMecanumDrive) {
             /// Don't forget to check the order of the motors
             mecanumDriveSubsystem = new MecanumDriveSubsystem(hardware.leftFrontMotor, hardware.rightFrontMotor, hardware.leftRearMotor, hardware.rightRearMotor, hardware.imu);
         }
 
         if (enableLift) {
-            liftSubsystem = new LiftSubsystem(hardware.liftMotor);
+            liftSubsystem = new LiftSubsystem(hardware.leftLiftMotor, hardware.rightLiftMotor, hardware::getVoltage);
```

Comment:
This probably shouldn't be passed as a supplier. The point of the voltage is to know at the *beginning* of the match if the battery is low. The voltage drops significantly while the lift is in operation. If we're dealing with a low battery, we should scale the PID values up, but you won't be able to get a useful read from the battery while the lift is running.